woocommerce/woocommerce
# Initial support to inline notices

### All Submissions:






### Changes proposed in this Pull Request:



This Pull Request enables initial support to inline notices, changing `wc_add_notice()` to accept an array of data (e.g. `['id' => 'billing_email']`), this data is displayed in the notice template (e.g `<ul class="woocommerce-error" role="alert"><li data-id="billing_email"> Billing Email is a required field.</li></ul>`), so in the future (or in 3rd party plugins or themes) will be possible to move those messages to each specific input in order to display inline notices.

### How to test the changes in this Pull Request:

1. Apply this PR
2. Try place an order missing to fill some fields or using an incorrect postcode (depends on the selected country to proper validate it, if you want to test it, you can test with US, UK, or Brazil)
3. Note that the notices display like before, but with the new HTML where includes `data-id`.

### Other information:




### Changelog entry

> Dev - Added initial support for inline notices on checkout.
